packages/vue-app/template/server.js

```diff
@@ -321,8 +321,7 @@ const render404Page = () => {
         promise.then((asyncDataResult) => {
           ssrContext.asyncData[Component.cid] = asyncDataResult
           applyAsyncData(Component)
-          return asyncDataResult
-        })
+        }).catch(() => {})
```

Comment:
removed `return` since this promise is not chained with anything so it doesn't need to return anything.